File: VHBoomMenuButtonDemo/NavigationBarVC.m
//
//  NavigationBarVC.m
//  VHBoomMenuButton
//
//  Created by Nightonke on 2017/4/14.
//  Copyright © 2017 Nightonke. All rights reserved.
//

#import "NavigationBarVC.h"
#import "BuilderManager.h"
#import <BoomMenuButton/BoomMenuButton.h>

@interface NavigationBarVC ()

@end

@implementation NavigationBarVC

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    VHBoomMenuButton *bmb1 = [[VHBoomMenuButton alloc] initWithFrame:CGRectMake(0, 0, 44, 44)];
    bmb1.buttonEnum = VHButtonTextOutsideCircle;
    bmb1.piecePlaceEnum = VHPiecePlaceDOT_9_1;
    bmb1.buttonPlaceEnum = VHButtonPlaceSC_9_1;
    bmb1.hasBackground = NO;
    bmb1.duration = 1;
    for (int i = 0; i < bmb1.pieceNumber; i++)
    {
        [bmb1 addBuilder:[BuilderManager textOutsideCircleButtonBuilderWithDifferentPieceColor]];
    }
    
    VHBoomMenuButton *bmb2 = [[VHBoomMenuButton alloc] initWithFrame:CGRectMake(44, 0, 44, 44)];
    bmb2.buttonEnum = VHButtonHam;
    bmb2.piecePlaceEnum = VHPiecePlaceHAM_4;
    bmb2.buttonPlaceEnum = VHButtonPlaceHAM_4;
    bmb2.hasBackground = NO;
    bmb2.duration = 1;
    bmb2.boomEaseName = VHEaseOutCubic;
    bmb2.reboomEaseName = VHEaseOutCubic;
    bmb2.boomEnum = VHBoomParabola2;
    for (int i = 0; i < bmb2.pieceNumber; i++)
    {
        [bmb2 addBuilder:[BuilderManager hamButtonBuilderWithDifferentPieceColor]];
    }
    
    UIBarButtonItem *negativeSpacer = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemFixedSpace target:nil action:nil];
    [negativeSpacer setWidth:-16];
    
    UIView *bmbs = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 88, 44)];
    [bmbs addSubview:bmb1];
    [bmbs addSubview:bmb2];
    
    self.navigationItem.rightBarButtonItems = @[negativeSpacer,
                                                [[UIBarButtonItem alloc] initWithCustomView:bmbs]];
}

@end

File: VHBoomMenuButtonSwiftDemo/NavigationBarVC.swift
//
//  NavigationBarVC.swift
//  VHBoomMenuButtonSwiftDemo
//
//  Created by Nightonke on 2017/5/4.
//  Copyright © 2017 Nightonke. All rights reserved.
//

import UIKit
import BoomMenuButton

class NavigationBarVC: BoomMenuButtonVC {

    override func viewDidLoad() {
        super.viewDidLoad()

        let bmb1: BoomMenuButton = BoomMenuButton.init(frame: CGRect.init(x: 0, y: 0, width: 44, height: 44))
        bmb1.buttonEnum = .textOutsideCircle
        bmb1.piecePlaceEnum = .dot_9_1
        bmb1.buttonPlaceEnum = .sc_9_1
        bmb1.hasBackground = false
        bmb1.duration = 1
        for _ in 0..<bmb1.piecePlaceEnum.pieceNumber() {
            bmb1.addBuilder(BuilderManager.textOutsideCircleButtonBuilderWithDifferentPieceColor())
        }
        
        let bmb2: BoomMenuButton = BoomMenuButton.init(frame: CGRect.init(x: 44, y: 0, width: 44, height: 44))
        bmb2.buttonEnum = .ham
        bmb2.piecePlaceEnum = .ham_4
        bmb2.buttonPlaceEnum = .ham_4
        bmb2.hasBackground = false
        bmb2.duration = 1
        bmb2.boomEaseName = Ease.outCubic
        bmb2.reboomEaseName = Ease.outCubic
        for _ in 0..<bmb2.piecePlaceEnum.pieceNumber() {
            bmb2.addBuilder(BuilderManager.hamButtonBuilderWithDifferentPieceColor())
        }
        
        let nagativeSpacer: UIBarButtonItem = UIBarButtonItem.init(barButtonSystemItem: UIBarButtonSystemItem.fixedSpace, target: nil, action: nil)
        nagativeSpacer.width = -16
        
        let bmbs: UIView = UIView.init(frame: CGRect.init(x: 0, y: 0, width: 88, height: 44))
        bmbs.addSubview(bmb1)
        bmbs.addSubview(bmb2)
        
        self.navigationItem.rightBarButtonItems = [nagativeSpacer, UIBarButtonItem.init(customView: bmbs)]
    }

}
